examples/Tests/EQDiagram.hpp

```diff
@@ -0,0 +1,267 @@
+#pragma once
+
+#include <DspFilters/Filter.h>
+#include <DspFilters/Cascade.h>
+#include <avnd/concepts/painter.hpp>
+#include <cmath>
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace utilities
+{
+float max(const float* values, int n)
+{
+  float max = *values;
+  for (auto i = 1; i < n; ++i)
+  {
+    if (max < *(values + i))
+      max = *(values + i);
+  }
+
+  return max;
+}
+
+float min(const float* values, int n)
+{
+  float min = *values;
+  for (auto i = 1; i < n; ++i)
+  {
+    if (min > *(values + i))
+      min = *(values + i);
+  }
+
+  return min;
+}
+
+float absolute(float val)
+{
+  if (val > 0)
+    return val;
+  return -val;
+}
+
+std::string to_string(float val, int precision)
+{
+  std::stringstream stream;
+  stream << std::fixed << std::setprecision(precision) << val;
+  return stream.str();
+}
+}
+
+/*  The following object handles the computation of the values to plot the Parametric EQ (Bode) diagram.
+    It needs two values to properly run:
+        - The number of plot values to compute (N) which is passed as a template argument;
+        - The sample rate (sampleRate), needed to normalize frequencies when computing a filter's response to a certain frequency, which should be set before use of the object.
+*/
+template <int N, int bandNb>
+struct EQDiagramCalculator
+{
+  EQDiagramCalculator()
+      : sampleRate{44100.f}
+  {
+    auto eps = (range::max - range::min) / (N - 1);
+    for (auto i = 0; i < N; ++i)
+      frequencies.push_back(range::min + eps * i);
+    filters.resize(bandNb);
+  }
+
+  struct range
+  {
+    static const int plotAmount = N;
+    static const double min = 20.f;
+    static const double max = 20000.f;
+  };
+
+  float sampleRate;
+
+  std::vector<float> frequencies{}; //abscissa values
+  std::vector<float> responses{};   //ordinate values
+
+  using FilterHolder = Dsp::Filter;
+  std::vector<const FilterHolder*> filters{};
+
+  void addFilter(int band, const FilterHolder* filter)
+  {
+    if (filters[band] != filter)
+    {
+      filters[band] = filter;
+    }
+  }
+
+  bool removeFilter(int band) 
+  { 
+    if(filters[band] == nullptr)
+      return false;
+    filters[band] = nullptr;
+    return true;
+  }
+  /*
+        We consider each filter to be connected in series. In this case, the resulting transfer function is the 
+        multiplication of each filter's transfer function and the gain response in the sum of the norms of 
+        each filter's complex responses.
+    */
+  void computeDiagram()
+  {
+    // we erase previous values contained in responses as there might be new filters added / removed filters since last computation
+    responses = std::vector<float>();
+    for (auto& w : frequencies)
+    {
+      float res{};
+      for (auto& f : filters)
+      {
+        if (f)
+          res += log10(std::norm(f->response(w / sampleRate)));
```

Comment:
std::log10 (also must take care to ensure that its argument is never zero as it would cause very bad noises)